### PR TITLE
Update Spring tool suite to latest version

### PR DESCRIPTION
**Pull request by StevenACoffman**
This is rushed, but I noticed the cask version available wouldn't work with tomcat 8, so just to make everything work on a fresh system, I updated this cask. I noticed some weirdness in that the STS URL has only the eclipse e4.4 in one place, but 4.4.2 in another, but it looks like it's always been that way and the cask takes care of that.

### CODE REVIEW

File: Casks/sts.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
   module Utils
     def self.based_on_eclipse
-      '4.3.2'   # find eclipse version at http://spring.io/tools/sts/all
+      '4.4.2'   # find eclipse version at http://spring.io/tools/sts/all
     end
   end
-
-  url "http://download.springsource.com/release/STS/#{version}/dist/e#{Utils.based_on_eclipse.gsub(/\.\d$/, '')}/spring-tool-suite-#{version}.RELEASE-e#{Utils.based_on_eclipse}-macosx-cocoa-x86_64-installer.dmg"
+  #http://dist.springsource.com/release/STS/3.6.4.RELEASE/dist/e4.4/spring-tool-suite-3.6.4.RELEASE-e4.4.2-macosx-cocoa-x86_64.tar.gz
```
Comment (vitorgalvao) [inline Casks/sts.rb]: Please remove this comment — it’s useless.

**Comment by vitorgalvao** [comment]
Please [squash your commits](http://davidwalsh.name/squash-commits-git).

**Comment by StevenACoffman** [comment]
Ok, removed comment, squashed commits, and now it should be ok.

File: Casks/sts.rb [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+  version '3.6.4'
+  sha256 '9627e702f83efa19d25f6041bd86a58395ba1cc769c6e76b3f628d6c23900633'
 
   module Utils
     def self.based_on_eclipse
-      '4.3.2'   # find eclipse version at http://spring.io/tools/sts/all
+      '4.4.2'   # find eclipse version at http://spring.io/tools/sts/all
     end
   end
-
```
Comment (vitorgalvao) [inline Casks/sts.rb]: This empty line shouldn’t be removed.

**Comment by vitorgalvao** [comment]
You’ve grabbed an unrelated commit in the process.

**Comment by StevenACoffman** [comment]
I'll close this and do it again better. Thanks.

**Pull request closed without merging**